server/sonar-server/src/main/java/org/sonar/server/webhook/ws/SearchAction.java

```diff
@@ -86,44 +87,63 @@ public void define(WebService.NewController controller) {
 
     try (DbSession dbSession = dbClient.openSession(true)) {
 
+      OrganizationDto organizationDto;
+      if (isNotBlank(organizationKey)) {
+        Optional<OrganizationDto> dtoOptional = dbClient.organizationDao().selectByKey(dbSession, organizationKey);
+        organizationDto = checkFoundWithOptional(dtoOptional, "No organization with key '%s'", organizationKey);
+      } else {
+        organizationDto = defaultOrganizationDto(dbSession);
+      }
+
       if (isNotBlank(projectKey)) {
-        Optional<ComponentDto> component = dbClient.componentDao().selectByKey(dbSession, projectKey);
-        checkFoundWithOptional(component, "project %s does not exist", projectKey);
-        userSession.checkComponentPermission(ADMIN, component.get());
-        return new ArrayList<>(settingsFinder.loadComponentSettings(dbSession,
-          ImmutableSet.of("sonar.webhooks.project"), component.get()).get(component.get().uuid()));
+
+        Optional<ComponentDto> optional = ofNullable(dbClient.componentDao().selectByKey(dbSession, projectKey).orNull());
+        ComponentDto componentDto = checkFoundWithOptional(optional, "project %s does not exist", projectKey);
+        webhookSupport.checkUserPermissionOn(componentDto);
+        webhookSupport.checkThatProjectBelongsToOrganization(componentDto, organizationDto, "Project '%s' does not belong to organisation '%s'", projectKey, organizationKey);
+        webhookSupport.checkUserPermissionOn(componentDto);
+        return dbClient.webhookDao().selectByProjectUuid(dbSession, componentDto.uuid());
+
       } else {
-        userSession.checkIsSystemAdministrator();
-        return settingsFinder.loadGlobalSettings(dbSession, ImmutableSet.of("sonar.webhooks.global"));
+
+        webhookSupport.checkUserPermissionOn(organizationDto);
+        return dbClient.webhookDao().selectByOrganizationUuid(dbSession, organizationDto.getUuid());
+
       }
+
     }
   }
 
-  private static void writeResponse(Request request, Response response, List<Setting> settings) {
+  private static void writeResponse(Request request, Response response, List<WebhookDto> webhookDtos) {
 
     Builder responseBuilder = newBuilder();
 
-    settings
+    webhookDtos
       .stream()
-      .map(Setting::getPropertySets)
-      .flatMap(Collection::stream)
-      .forEach(map -> responseBuilder.addWebhooksBuilder()
-        .setKey("")
-        .setName(map.get("name"))
-        .setUrl(map.get("url")));
+      .forEach(webhook -> responseBuilder.addWebhooksBuilder()
+        .setKey(webhook.getUuid())
+        .setName(webhook.getName())
+        .setUrl(webhook.getUrl()));
 
     writeProtobuf(responseBuilder.build(), request, response);
   }
 
+  private OrganizationDto defaultOrganizationDto(DbSession dbSession) {
+    String uuid = defaultOrganizationProvider.get().getUuid();
+    Optional<OrganizationDto> organizationDto = dbClient.organizationDao().selectByUuid(dbSession, uuid);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this fully qualified name with "Optional" ([squid:S1942](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWF2DiAldoZs8euTWtP7&open=AWF2DiAldoZs8euTWtP7)

[ISSUE_KEY]: AWF2DiAldoZs8euTWtP7